$MirOS: ports/net/djbdns/patches/patch-src_dns_random_c,v 1.1 2006/01/17 21:32:17 tg Exp $

Sorry, Dan.

--- src/dns_random.c.orig	Sun Feb 11 21:11:23 2001
+++ src/dns_random.c	Mon Oct 27 21:35:45 2008
@@ -1,3 +1,4 @@
+#include <stdlib.h>
 #include <unistd.h>
 #include "dns.h"
 #include "taia.h"
@@ -46,7 +47,8 @@ void dns_random_init(const char data[128
 
   in[8] = getpid();
   in[9] = getppid();
-  /* more space in 10 and 11, but this is probably enough */
+  in[10] = arc4random();
+  in[11] = arc4random();
 }
 
 unsigned int dns_random(unsigned int n)
